src/Client/ClientBaseHelpers.cpp

```diff
@@ -237,4 +238,25 @@ void highlight(const String & query, std::vector<replxx::Replxx::Color> & colors
 }
 #endif
 
+String formatQuery(String query)
+{
+    bool multiline_query = query.contains('\n');
+    ParserQuery parser(query.data() + query.size(), /*allow_settings_after_format_in_insert_=*/ false, /*implicit_select_=*/ false);
+    const ASTPtr ast = parseQuery(parser, query, /*max_query_size=*/ 0, /*max_parser_depth=*/ 0, /*max_parser_backtracks=*/ 0);
+
+    bool insert_with_data = false;
+    if (auto * insert_query = ast->as<ASTInsertQuery>(); insert_query && insert_query->hasInlinedData())
+        insert_with_data = true;
```

Comment:
Why not to simply format this? If the query was parsed then it should be able to format into string with the inlined data I suppose, otherwise it should throw an exception in parser